### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by itexpert120**

### CODE REVIEW

File: apps/builddao/widget/components/User.jsx [hunk 1]
```diff
@@ -58,8 +58,9 @@ const Overlay = (props) => (
   <Link
     className="link-dark text-truncate d-inline-flex mw-100"
     to={href({
-      widgetSrc: "mob.near/widget/ProfilePage",
+      widgetSrc: "mob.near/widget/app",
```
Comment (elliotBraem) [inline apps/builddao/widget/components/User.jsx:61]: mob.near should be buildhub.near

File: apps/builddao/widget/components/buttons/UserDropdown.jsx [hunk 2]
```diff
@@ -181,7 +181,7 @@ return (
         <Link
           className="dropdown-item"
           type="button"
-          to={`/mob.near/widget/ProfilePage?accountId=${context.accountId}`}
+          to={`/buildhub.near/widget/app?page=profile&accountId=${context.accountId}`}
```
Comment (elliotBraem) [inline apps/builddao/widget/components/buttons/UserDropdown.jsx:184]: Use href

File: apps/builddao/widget/components/profile/AccountFeed.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+const commentAccounts = props.commentAccounts;
+const renderedPosts = {};
+
+const makePostItem = (a) => ({
+  type: "social",
+  path: `${a.accountId}/post/main`,
+  blockHeight: a.blockHeight,
+});
+
+const renderPost = (a) => {
```
Comment (elliotBraem) [inline apps/builddao/widget/components/profile/AccountFeed.jsx:44]: This all could be replaced with devs.near/widget/Feed module if you want, just set index to post/main, repost/main

File: apps/builddao/widget/components/User.jsx (hunk 1 above)
Comment (itexpert120) [reply apps/builddao/widget/components/User.jsx:61]: Oh, how did I miss it lol

File: apps/builddao/widget/components/profile/ProfileLine.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    )}
+  </>
+);
+
+inner = link ? (
+  <Link
+    href={
+      link !== true
+        ? link
+        : `/buildhub.near/widget/app?page=profile&accountId=${accountId}`
```
Comment (elliotBraem) [inline apps/builddao/widget/components/profile/ProfileLine.jsx:44]: href

File: apps/builddao/widget/components/profile/WidgetMetadata.jsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+              data-bs-toggle="collapse"
+              data-bs-target={`#${descriptionKey}`}
+              aria-expanded="false"
+              aria-controls={descriptionKey}
+            >
+              <i className="bi bi-arrows-angle-expand me-1"></i>Show details
+            </button>
+          )}
+          <Link
+            href={`/mob.near/widget/WidgetSource?src=${widgetPath}`}
```
Comment (elliotBraem) [inline apps/builddao/widget/components/profile/WidgetMetadata.jsx:132]: href

File: apps/builddao/widget/page/projects.jsx [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
 const { theme } = VM.require("buildhub.near/widget/config.theme") ?? {
-  theme: {}
+  theme: {},
 };
 
 const { SidebarLayout } = VM.require(
   "buildhub.near/widget/template.SidebarLayout"
 ) || {
-  SidebarLayout: () => <>Layout loading...</>
+  SidebarLayout: () => <>Layout loading...</>,
```
Comment (elliotBraem) [inline apps/builddao/widget/page/projects.jsx:14]: Can we replace with empty loading? <></>

File: apps/builddao/widget/components/profile/AccountFeed.jsx (hunk 3 above)
Comment (itexpert120) [reply apps/builddao/widget/components/profile/AccountFeed.jsx:44]: I think I'll leave it as is...

File: apps/builddao/widget/Compose.jsx [hunk 7]
```diff
@@ -157,6 +155,7 @@ const postToCustomFeed = ({ feed, text, labels }) => {
       setPostContent("");
       Storage.privateSet(draftKey, props.template || "");
       setHandler("autocompleteSelected"); // this is a hack to force the iframe to update
+      setComposeKey(generateUID());
```
Comment (elliotBraem) [inline apps/builddao/widget/Compose.jsx:158]: How come this? How would you remember the uuid of your draft to revisit it?

Comment (itexpert120) [reply apps/builddao/widget/Compose.jsx:158]: Oh, this is just there to trigger a rerender of the compose iframe. Usually the iframe does not rerender and causes the iframe to still contain the post content after posting. This fixes that

Comment (itexpert120) [reply apps/builddao/widget/Compose.jsx:158]: The uuid and drafts are different. The UUID is for the iframe only just to trigger a rerender

File: apps/builddao/widget/Resources.jsx [hunk 8]
```diff
@@ -1,7 +1,9 @@
-const { MarkdownView } =
-  VM.require("buildhub.near/widget/md-view") || (() => <></>);
-const { Button } =
-  VM.require("buildhub.near/widget/components.Button") || (() => <></>);
+const { MarkdownView } = VM.require("buildhub.near/widget/md-view") || {
```
Comment (elliotBraem) [inline apps/builddao/widget/Resources.jsx:1]: But this one is perfect, nicd

File: apps/builddao/widget/components/Bullet.jsx [hunk 9]
```diff
@@ -1,5 +1,6 @@
-const { theme } =
```
Comment (elliotBraem) [inline apps/builddao/widget/components/Bullet.jsx]: Any way we can remove this coming from themeprovider/ theme?

File: apps/builddao/widget/Proposals.jsx [hunk 10]
```diff
@@ -1,11 +1,14 @@
 const { Button } = VM.require("buildhub.near/widget/components.Button") || {
   Button: <></>,
 };
-const DaoSDK = VM.require("sdks.near/widget/SDKs.Sputnik.DaoSDK") || (() => {});
+const DaoSDK = VM.require("sdks.near/widget/SDKs.Sputnik.DaoSDK") || {
```
[the remark this answers is not in the document]
Comment (itexpert120) [reply apps/builddao/widget/Proposals.jsx]: Just realized this. Fixed

File: apps/builddao/widget/components/Bullet.jsx (hunk 9 above)
Comment (itexpert120) [reply apps/builddao/widget/components/Bullet.jsx]: Well, it is an object and not a wrapper, else we could have wrapped all of our app within it

**Pull request merged**